src/main/java/com/jcabi/xml/ClasspathInput.java

```diff
@@ -119,16 +111,30 @@ public String getEncoding() {
     }
 
     @Override
+    @SuppressWarnings("PMD.DoNotThrowExceptionInFinally")
     public String getStringData() {
+        final InputStream stream = getClass()
+                                 .getResourceAsStream(this.systemid);
         try {
-            return IOUtils.toString(
-                this.stream,
+            final String data = IOUtils.toString(
+                stream,
                 Charset.forName("UTF-8")
             );
+            return data;
         } catch (final IOException exception) {
             throw new IllegalArgumentException(
                 "Unable to read input", exception
             );
+        } finally {
+            if (stream != null) {
```

Comment:
@amihaiemil `stream` should never be `null` here. instead, check for null above, right after the call to `getResourceAsStream`